### PR TITLE
JAMES-2906 Add update flag test for ES

### PR DESCRIPTION
**Pull request by Arsnael**
Related to #2745 

We actually lacked a flag update test on ES. Works on master but definitely not in the current POC _source removal. But should add it anyway !

### CODE REVIEW

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/search/AbstractMessageSearchIndexTest.java [hunk 1]
```diff
@@ -520,6 +520,22 @@ public void flagIsSetShouldReturnUidOfMessageMarkedAsSeenWhenUsedWithFlagSeen()
         assertThat(messageSearchIndex.search(session, mailbox, searchQuery))
             .containsOnly(m6.getUid());
     }
+
+    @Test
+    public void flagUpdateShouldReturnUidOfMessagesMarkedAsSeenAfterMessageUpdatedWithFlagSeen() throws MailboxException {
```
Comment (chibenwa) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/search/AbstractMessageSearchIndexTest.java]: searchShouldReturnSeenMessagesWhenFlagsGotUpdated

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/search/AbstractMessageSearchIndexTest.java [hunk 2]
```diff
@@ -520,6 +520,22 @@ public void flagIsSetShouldReturnUidOfMessageMarkedAsSeenWhenUsedWithFlagSeen()
...
+            MessageManager.FlagsUpdateMode.ADD,
+            MessageRange.one(m5.getUid()),
+            session);
+
+        await();
+
+        SearchQuery searchQuery = new SearchQuery(SearchQuery.flagIsSet(Flags.Flag.SEEN));
+
+        assertThat(messageSearchIndex.search(session, mailbox, searchQuery))
+            .containsOnly(m5.getUid(), m6.getUid());
```
Comment (mbaechler) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/search/AbstractMessageSearchIndexTest.java]: we don't care about m6, do we?

**Comment by chibenwa** [comment]
test this please

**Comment by Arsnael** [comment]
As I understood seems there was a restart yesterday...

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**